### PR TITLE
Restore 2.7 compatibility as well as Python 3.12.3 and httpd 2.4.59.

### PR DESCRIPTION
**Pull request by grisha**
This PR fixes all the recent test failures.

### CODE REVIEW

**Comment by grisha** [comment]
@dviererbe @panlinux @Maxime2 would appreciate it if you could verify that this PR works for you. 

For whatever reason my local checkout was behind and I ended up repeating some things that were already done such as the + in the version, sorry about that.

With this PR I have all the tests passing using various combinations of Pythons 2.7.18 and 3.12.3 and httpd 2.4.59 as well as 2.2.15. 

I'm not sure how valuable 2.7 compatibility is, but people do notice when it breaks apparently. 

Also it looks like 3.12 has deprecated setup.py / easy_install, but they're still around so for now this still works.

**Comment by Maxime2** [comment]
@grisha On MacBook Pro with macOS Sonoma 14.3.1 and Python 3.11.8 it fails two tests, due to [a bug in Apache2](https://github.com/apache/httpd/pull/428):
```
======================================================================
FAIL: test_publisher_auth_nested (__main__.PerRequestTestCase.test_publisher_auth_nested)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/maxime/git/mod_python/test/test.py", line 2504, in test_publisher_auth_nested
    self.fail(repr(rsp))
AssertionError: b'<!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML 2.0//EN">\n<html><head>\n<title>400 Bad Request</title>\n</head><body>\n<h1>Bad Request</h1>\n<p>Your browser sent a request that this server could not understand.<br />\n</p>\n</body></html>\n'

======================================================================
FAIL: test_publisher_auth_method_nested (__main__.PerRequestTestCase.test_publisher_auth_method_nested)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/maxime/git/mod_python/test/test.py", line 2534, in test_publisher_auth_method_nested
    self.fail(repr(rsp))
AssertionError: b'<!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML 2.0//EN">\n<html><head>\n<title>400 Bad Request</title>\n</head><body>\n<h1>Bad Request</h1>\n<p>Your browser sent a request that this server could not understand.<br />\n</p>\n</body></html>\n'

----------------------------------------------------------------------
Ran 71 tests in 9.358s

FAILED (failures=2)
```

**Comment by Maxime2** [comment]
On Ubuntu 22.04 with Python 3.10.12 and Apache2 fixed for that bug it passes all tests successfully:
```
Ran 72 tests in 10.079s

OK
```
Though it has one warning:
```
* Testing C/Py version mismatch warning
  Creating config....
    listen port: 43651
  Starting Apache....
     /usr/sbin/apache2  -k start -f /home/maxime/git/mod_python/test/conf/test.conf
  Stopping Apache...
     /usr/sbin/apache2 -k stop -f /home/maxime/git/mod_python/test/conf/test.conf
/home/maxime/git/mod_python/test/test.py:2890: DeprecationWarning: The distutils package is deprecated and slated for removal in Python 3.12. Use setuptools or check PEP 632 for potential alternatives
  from distutils.sysconfig import get_python_lib
/home/maxime/git/mod_python/test/test.py:2890: DeprecationWarning: The distutils.sysconfig module is deprecated, use sysconfig instead
  from distutils.sysconfig import get_python_lib
.
```

**Comment by Maxime2** [comment]
@grisha I have tried with `0x030b0000` and got all test passed successfully.

Though I needed to uninstall previous version using pip to have correct version picked up. That mean putting git commit hash into version string is not a good idea for correct versioning, perhaps the date of that commit in the format YYYYMMDD would work. Also would be useful to have `uninstall` target in the makefile, or have uninstall instructions in the documentation.

**Comment by grisha** [comment]
> @grisha I have tried with 0x030b0000 and got all test passed successfully.

Wonderful!

> Though I needed to uninstall previous version using pip to have correct version picked up. That mean putting git commit hash into version string is not a good idea for correct versioning

I'll check it out, I believe this will resolve itself once there is a git tag - version.sh should not append the hash in this case.

File: configure [hunk 1]
```diff
@@ -1,10 +1,9 @@
 #! /bin/sh
 # Guess values for system-dependent variables and create Makefiles.
-# Generated by GNU Autoconf 2.71.
+# Generated by GNU Autoconf 2.69.
```
[the remark this answers is not in the document]
Comment (grisha) [reply configure:3]: Not really.

**Pull request merged**